flow/otel_metrics/otel_manager.go

```diff
@@ -0,0 +1,54 @@
+package otel_metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+)
+
+type OtelManager struct {
+	MetricsProvider    *sdkmetric.MeterProvider
+	Meter              metric.Meter
+	Float64GaugesCache map[string]*Float64Gauge
+	Int64GaugesCache   map[string]*Int64Gauge
+}
+
+// newOtelResource returns a resource describing this application.
+func newOtelResource(otelServiceName string) (*resource.Resource, error) {
+	r, err := resource.Merge(
+		resource.Default(),
+		resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(otelServiceName),
+		),
+	)
+
+	return r, err
+}
+
+func SetupOtelMetricsExporter() (*sdkmetric.MeterProvider, error) {
```

Comment:
can modify to accept the service name (`flow-worker`) to make it service agnostic (and reusable across `api`, `worker`, `snapshot`).

Reply:
changed